pyethapp/accounts.py

```diff
@@ -480,9 +480,7 @@ def get_by_address(self, address):
         assert len(address) == 20
         accounts = [account for account in self.accounts if account.address == address]
         if len(accounts) == 0:
-            raise KeyError('account not found by address', address=address.encode('hex'))
-        elif len(accounts) > 1:
-            log.warning('multiple accounts with same address found', address=address.encode('hex'))
```

Comment:
OK I've reverted it. I wanted to covert it with a test, but it requires us to introduce pytest mocks.
So it will be for next time.